components/org.wso2.carbon.business.rules.web/src/components/BusinessRuleFromScratchForm.js

```diff
@@ -0,0 +1,754 @@
+import BusinessRulesMessages from "../utils/BusinessRulesMessages";
+import ProgressDisplay from "./ProgressDisplay";
+import Grid from 'material-ui/Grid';
+import Paper from 'material-ui/Paper';
+
+/**
+ * Represents a form, shown to for Business Rules from scratch
+ */
+
+// Styles related to this component
+const styles = {
+    addFilterRuleButton: {
+        backgroundColor: '#EF6C00',
+        color: 'white'
+    },
+    button: {
+        backgroundColor: '#EF6C00',
+        color: 'white',
+        marginRight: 10
+    },
+    secondaryButton: {
+        marginRight: 10
+    },
+    paper: {
+        padding: 40,
+        paddingTop: 15,
+        paddingBottom: 15,
+    },
+    root: {
+        width: '100%',
+        maxWidth: 360,
+        position: 'relative',
+        overflow: 'auto',
+        maxHeight: 300,
+    },
+    rootGrid: {
+        flexGrow: 1,
+        paddingTop: 20
+    },
+    propertyComponentPadding: {
+        paddingLeft: 40,
+        paddingRight: 40
+    },
+    listSection: {
+        background: 'inherit',
+    },
+    formRoot: {
+        flexGrow: 1,
+        marginTop: 30,
+    },
+    formPaper: {
+        padding: 50,
+        maxWidth: 1000,
+    }
+}
+
+class BusinessRuleFromScratchForm extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            formMode: props.formMode,
+            templateGroups: props.templateGroups,
+            selectedTemplateGroup: props.selectedTemplateGroup,
+            inputRuleTemplates: props.inputRuleTemplates,
+            outputRuleTemplates: props.outputRuleTemplates,
+
+            // Present only in 'edit' mode
+            businessRuleName: props.businessRuleName,
+            businessRuleUUID: props.businessRuleUUID,
+            // Rule Templates, whose properties will be used to generate form
+            selectedInputRuleTemplate: props.selectedInputRuleTemplate,
+            selectedOutputRuleTemplate: props.selectedOutputRuleTemplate,
+
+            // To store values given for properties displayed in the form
+            businessRuleProperties: props.businessRuleProperties,
+
+            // Expanded states of components
+            isInputComponentExpanded: false,
+            isFilterComponentExpanded: false,
+            isOutputComponentExpanded: false,
+
+            // For displaying messages / errors
+            displayDialog: false,
+            dialogTitle: '',
+            dialogContentText: '',
+            dialogPrimaryButtonText: ''
+        }
+
+        // Assign default values of properties as entered values in create mode
+        if (this.state.formMode === BusinessRulesConstants.BUSINESS_RULE_FORM_MODE_CREATE) {
+            // Create mode
+            let state = this.state
+            // Set an empty business rule values in the state
+            state['selectedInputRuleTemplate'] = {}
+            state['selectedOutputRuleTemplate'] = {}
+            state['businessRuleName'] = ''
```

Comment:
do not use state[""]. It is better to use this.setState ({
    businessRuleName: {}
})

So you do no need to do following,

let state = this.state
// assigning values state
this.state = state